### PR TITLE
Sgl/flex porting

### PR DESCRIPTION
**Pull request by sgliner-ledger**

### CODE REVIEW

File: src/ui/display_nbgl.c [hunk 1]
```diff
@@ -44,276 +30,95 @@ static void ux_flow_response_true(void) {
...
-    } else {
-        ux_flow_response_false();
-        nbgl_useCaseStatus(transactionContext.rejected_status, false, ui_menu_main);
-    }
-}
-
 static void status_cancel(void) {
-    status_confirmation_callback(false);
+    nbgl_useCaseStatus(rejected_status, false, ui_menu_main);
+    ux_flow_response_false();
```
Comment (xchapron-ledger) [inline src/ui/display_nbgl.c:36]: check if should'nt be the other way?

File: src/ui/display_nbgl.c [hunk 2]
```diff
@@ -44,276 +30,95 @@ static void ux_flow_response_true(void) {
...
-        nbgl_useCaseStatus(transactionContext.confirmed_status, true, ui_menu_main);
-    } else {
-        ux_flow_response_false();
-        nbgl_useCaseStatus(transactionContext.rejected_status, false, ui_menu_main);
-    }
-}
-
 static void status_cancel(void) {
-    status_confirmation_callback(false);
+    nbgl_useCaseStatus(rejected_status, false, ui_menu_main);
```
Comment (xchapron-ledger) [inline src/ui/display_nbgl.c:35]: Should be nbgl_useCaseReviewStatus ?

File: src/ui/display_nbgl.c [hunk 3]
```diff
@@ -44,276 +30,95 @@ static void ux_flow_response_true(void) {
...
-                                      .quitToken = CANCEL_TOKEN,
-                                      .tuneId = TUNE_TAP_CASUAL};
-
-    nbgl_pageContent_t content = {.type = TAG_VALUE_LIST,
-                                  .tagValueList.nbMaxLinesForValue = 8,
-                                  .tagValueList.nbPairs = transactionContext.tagValueList.nbPairs,
-                                  .tagValueList.pairs = transactionContext.tagValuePair};
+void ui_display_transaction_prompt(void) {
+    confirmed_status = "TRANSACTION\nSIGNED";
+    rejected_status = "Transaction rejected";
```
Comment (xchapron-ledger) [inline src/ui/display_nbgl.c:115]: to drop using nbgl_useCaseReviewStatus

File: src/ui/display_nbgl.c [hunk 4]
```diff
@@ -322,347 +127,460 @@ void ui_display_output_address_amount_flow(int index) {
...
+    // Setup list
+    pairList.nbMaxLinesForValue = 0;
+    pairList.nbPairs = 2;
+    pairList.pairs = pairs;
 
-    transactionContext.tagValueList.nbPairs = 2;
+    contentList[0].type = CENTERED_INFO;
+    contentList[0].content.centeredInfo.icon = &C_Bitcoin_64px;
+    contentList[0].content.centeredInfo.text1 = "Receive\nin known wallet";
+    contentList[0].content.centeredInfo.text2 = NULL;
```
Comment (xchapron-ledger) [inline src/ui/display_nbgl.c:227]: All functions calling nbgl_useCaseGenericReview could be factorized in one with only the pairList as parameter and maybe two strings.

File: src/ui/display_nbgl.c [hunk 5]
```diff
@@ -322,347 +127,460 @@ void ui_display_output_address_amount_flow(int index) {
...
+    genericContent.nbContents = 4;
+
+    nbgl_useCaseGenericReview(&genericContent, "Cancel", status_cancel);
 }
 
 // Continue light processing callback
 void ui_display_register_wallet_flow(void) {
-    transactionContext.tagValuePair[0].item = "Name";
-    transactionContext.tagValuePair[0].value = g_ui_state.wallet.wallet_name;
+    confirmed_status = "WALLET\nREGISTERED";
```
Comment (xchapron-ledger) [inline src/ui/display_nbgl.c:349]: should be lower case.

File: src/ui/display_nbgl.c [hunk 6]
```diff
@@ -322,347 +127,460 @@ void ui_display_output_address_amount_flow(int index) {
...
-    transactionContext.rejected_status = "Public key rejected";
-    nbgl_useCaseReviewStart(&C_Bitcoin_64px,
-                            "Confirm public key",
-                            "",
-                            "Cancel",
-                            suspicious_pubkey_warning,
-                            status_cancel);
+void ui_display_pubkey_suspicious_flow(void) {
+    confirmed_status = "PUBLIC KEY\nAPPROVED";
+    rejected_status = "Public key rejected";
```
Comment (xchapron-ledger) [inline src/ui/display_nbgl.c:297]: To be replaced with nbgl_useCaseReviewStatus calling STATUS_TYPE_ADDRESS_VERIFIED / STATUS_TYPE_ADDRESS_REJECTED?

File: src/ui/menu_nbgl.c [hunk 7]
```diff
@@ -21,41 +21,41 @@
...
-    nbgl_useCaseSettings("Bitcoin", 0, 1, false, ui_menu_main, navigation_cb, NULL);
-}
-
-void ui_menu_about_testnet(void) {
-    nbgl_useCaseSettings("Bitcoin Testnet", 0, 1, false, ui_menu_main, navigation_cb, NULL);
-}
-
 void ui_menu_main_flow_bitcoin(void) {
-    nbgl_useCaseHome("Bitcoin", &C_Bitcoin_64px, NULL, false, ui_menu_about, exit);
+    nbgl_useCaseHomeAndSettings("Bitcoin",
```
Comment (xchapron-ledger) [inline src/ui/menu_nbgl.c:39]:
```suggestion
    nbgl_useCaseHomeAndSettings(APPNAME,
```
?

File: src/ui/menu_nbgl.c [hunk 8]
```diff
@@ -21,41 +21,41 @@
...
+                                &C_Bitcoin_64px,
+                                NULL,
+                                INIT_HOME_PAGE,
+                                NULL,
+                                &infoList,
+                                NULL,
+                                exit);
 }
 
 void ui_menu_main_flow_bitcoin_testnet(void) {
```
Comment (xchapron-ledger) [inline src/ui/menu_nbgl.c:49]: Really needed?

**Review by xchapron-ledger — DISMISSED** [review]

File: src/ui/menu_nbgl.c (hunk 8 above)
Comment (sgliner-ledger) [reply src/ui/menu_nbgl.c:49]: The appname is set to `Bitcoin Test` and we want the homescreen to read bitcoin testnet

Comment (xchapron-ledger) [reply src/ui/menu_nbgl.c:49]: But this will be done automatically now that `ui_menu_main_flow_bitcoin` uses `APPNAME`.

**Review by tdejoigny-ledger — DISMISSED** [review]

**Review by tdejoigny-ledger — APPROVED** [review]

**Pull request merged**